app/modules/phonebook/endpoints_phonebook.py

```diff
@@ -0,0 +1,480 @@
+    status_code=204,
+)
+async def update_membership(
+    membership: schemas_phonebook.MembershipEdit,
+    membership_id: str,
+    user: models_core.CoreUser = Depends(is_user_a_member),
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Update a Membership.
+
+    **This endpoint is only usable by CAA, BDE and association's president**
+    """
+
+    membership_db = await cruds_phonebook.get_membership_by_id(
+        membership_id=membership_id, db=db
+    )
+    if not membership_db:
+        raise HTTPException(
+            status_code=400,
+            detail=f"No membership to update for membership_id {membership_id}",
+        )
+
+    if not (
+        is_user_member_of_an_allowed_group(
+            user=user, allowed_groups=[GroupType.CAA, GroupType.BDE]
+        )
+        or await cruds_phonebook.is_user_president(
+            association_id=membership_db.association_id, user=user, db=db
+        )
+    ):
+        raise HTTPException(
+            status_code=403,
+            detail=f"You are not allowed to update membership for association {membership_db.association_id}",
+        )
+
+    membership_edit = schemas_phonebook.MembershipEdit(**membership.model_dump())
+    await cruds_phonebook.update_membership(membership_edit, membership_id, db)
+
+
+@module.router.delete(
+    "/phonebook/associations/memberships/{membership_id}",
+    status_code=204,
+)
+async def delete_membership(
+    membership_id: str,
+    user: models_core.CoreUser = Depends(is_user_a_member),
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Delete a membership.
+
+    **This endpoint is only usable by CAA, BDE and association's president**
+    """
+
+    membership = await cruds_phonebook.get_membership_by_id(membership_id, db)
+    if not membership:
+        raise HTTPException(
+            status_code=400,
+            detail=f"No membership to delete for membership_id {membership_id}",
+        )
+
+    if not (
+        is_user_member_of_an_allowed_group(
+            user=user, allowed_groups=[GroupType.CAA, GroupType.BDE]
+        )
+        or await cruds_phonebook.is_user_president(
+            association_id=membership.association_id, user=user, db=db
+        )
+    ):
+        raise HTTPException(
+            status_code=403,
+            detail=f"You are not allowed to delete membership for association {membership.association_id}",
+        )
+
+    await cruds_phonebook.delete_membership(membership_id, db)
+
+
+# ---------------------------------------------------------------------------- #
+#                                     Logos                                    #
+# ---------------------------------------------------------------------------- #
+@module.router.post(
+    "/phonebook/associations/{association_id}/picture",
+    # response_model=standard_responses.Result,
+    status_code=201,
+)
+async def create_association_logo(
+    association_id: str,
+    image: UploadFile = File(...),
```

Comment:
This is not required anymore https://github.com/tiangolo/fastapi/releases/tag/0.78.0